File: week_02/codewars/chSung/200831_sum_of_digits_digital_root.java
public class DRoot {
    public static int digital_root(int num) {
        int sum = 0;
        while (num != 0) {
            sum += num % 10;
            num /= 10;
        }
        if (sum < 10) {
            return sum;
        } else {
            return digital_root(sum);
        }
    }
}

File: week_02/codewars/thkwon/200831_sum_of_digits_digital_root.py
import unittest


def digital_root(number):
    str_number = str(number)

    if number < 10:
        return number

    while len(str_number) > 1:

        sum_of_num = sum(int(element) for element in str_number)

        if sum_of_num >= 10:
            str_number = str(sum_of_num)
        else:
            return sum_of_num


class TestDigitalRoot(unittest.TestCase):
    def test_digital_root(self):
        self.assertEqual(digital_root(16), 7)

    def test_digital_root_number_is_0(self):
        self.assertEqual(digital_root(0), 0)

    def test_digital_root_number_should_return_given_number_when_number_is_less_than_10(self):
        self.assertEqual(digital_root(1), 1)
        self.assertEqual(digital_root(2), 2)
        self.assertEqual(digital_root(3), 3)
        self.assertEqual(digital_root(4), 4)
        self.assertEqual(digital_root(5), 5)
